cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -223,8 +223,13 @@ data ProjectBuildContext = ProjectBuildContext {
 runProjectPreBuildPhase
     :: Verbosity
     -> ProjectBaseContext
-    -> (ElaboratedInstallPlan -> IO ElaboratedInstallPlan)
-    -> IO ProjectBuildContext
+    -- ^ A function that selects a subset of the plan and returns it,
+    -- It can also return some arbitrary data, which can be useful in case
+    -- the caller needs it.
+    -> (ElaboratedInstallPlan -> IO (ElaboratedInstallPlan, a))
```

Comment:
Isn't `new-test` already done, according to that issue (also, it works for me)?

Or we have to redo it, since the current one relies on the "old" Setup, as most of the new-commands do (I think. They seem to do so.)?

`new-ench` builds but does not run the benchmark, same as run before my pr.